keps/sig-windows/0000-20181221-windows-group-managed-service-accounts-for-container-identity.md

```diff
@@ -99,105 +101,167 @@ A developer has a Windows web app that they would like to deploy in a container
+
+  The ability to specify credspecs for each container within a pod aligns with how security attributes like `runAsGroup`, `runAsUser`, etc. can be specified at the pod level and overridden at the container level if desired.
+
+6. A mutating webhook admission controller, GMSAAuthorizer, will act on pod creations and execute the following checks and actions:
+
+  - Check the pod spec for references to names of GMSACredSpec objects in the pod and per-container annotations [`pod.alpha.kubernetes.io/windows-gmsa-credspec-name` and `container.windows-gmsa-credspec-name.alpha.kubernetes.io/container-name` for Alpha] or securityContext fields [`podspec.securityContext.windows.gmsaCredSpecName` and `podspec.container[i].securityContext.windows.gmsaCredSpecName` Beta onwards]. Next, make sure the GMSACredSpec objects (referred to by name) do exist and prepare a JSON representation of the contents of the `credspec` field of the GMSACredSpec. If the specified names of GMSACredSpec cannot be resolved, pod creation will be failed by GMSAAuthorizer with 404: NotFound.
+
+  - Check the pod spec for inlined credspec JSON in the pod and per-container annotations [`pod.alpha.kubernetes.io/windows-gmsa-credspec` and `container.windows-gmsa-credspec.alpha.kubernetes.io/container-name` for Alpha] or securityContext fields [`podspec.securityContext.windows.gmsaCredSpec` and `podspec.container[i].securityContext.windows.gmsaCredSpec` Beta onwards]. These may exist as a result of the app admin optionally specifying the credspec JSON (besides the name).
+    - If a credspec JSON is not specified already, GMSAAuthorizer will look up the corresponding GMSACredSpec name and use the credspec JSON prepared from the previous step to populate the annotations or securityContext fields for credspec JSON.
+    - If a credspec JSON is specified, look up the corresponding GMSACredSpec name and perform a deep equal check of the members of the specified credspec JSON and prepared credspec JSON from previous step. If the deep equal check fails, GMSAAuthorizer will be fail pod creation with 400: BadRequest.
+
+  - Check the specified ServiceAccount is authorized for the `use` verb on all specified GMSACredSpecs at the pod level and for each container. If the authorization check fails due to absence of requisite RBAC roles, the pod creation will be failed with a 403: Forbidden. 
```

Comment:
if RBAC is not enabled, i think we should also fail with 403: Forbidden (if that's not already taken care of)

Reply:
Today, if RBAC and PSP is not enabled, starting a pod with privileged containers or extra capabilities is not failed/forbidden (please correct me if I am wrong). I think we should align with this existing model for gMSA and leave it up to the kubernetes cluster administrator (or distro vendors' setup mechanisms/cluster setup script authors) to configure and enable RBAC and the GMSAAuthorizer webhook if they want policy and authorization around gMSAs and service accounts in their clusters. Thoughts on keeping the core non opinionated in the context of gMSA and policy/authorization?

Reply:
I added it as a topic to discuss in our SIG-windows meet up today. i am torn half way on this, but i generally now that i think about this some more agree with being non-opinionated. in our docs we can stress the importance of RBAC in this context

Reply:
Based on discussion in SIG Windows, the conclusion was to stay non opinionated in this regard i.e. not enforce enablement of RBAC. This also aligns with how privileged pods and similar concepts are allowed without PSP/RBAC.